app/shared/helpers/array.js

```diff
@@ -118,4 +118,28 @@ _array_contains_by_properties = function(array, properties, object) {
 		}
 	}
 	return false;
+};
+
+// Will return a matching object by specified properties, otherwise null
+_array_find_by_properties = function(array, properties, object) {
```

Comment:
This is fine but if we already are pulling in Underscore, consider http://underscorejs.org/#where